readonly/load_models.py

```diff
@@ -86,59 +108,136 @@ def clone_git_model(project: dict[str, str]) -> None:
-        name = next(project["location"].glob("*.aird")).stem
-        natures = {
-            "library": "org.polarsys.capella.library.nature",
-            "project": "org.polarsys.capella.project.nature",
-        }
-
-        if project.get("nature", "") in natures:
-            nature = natures[project["nature"]]
-        else:
-            nature = natures["project"]
-
-        xml = E.projectDescription(
-            E.name(name),
-            E.comment(),
-            E.projects(),
-            E.buildSpec(),
-            E.natures(E.nature(nature)),
+    if project_description_file.exists():
+        project["etree"] = etree.fromstring(
+            project_description_file.read_bytes()
         )
+    else:
+        project["etree"] = generate_project_etree(project)
+
+
+def derive_project_name(project: _ProjectDict) -> None:
+    project["name"] = project["etree"].find("name").text.strip()
 
-        project_description_file.write_bytes(ET.tostring(xml))
+
+def generate_project_etree(project: _ProjectDict) -> etree.ElementTree:
+    # More information:
+    # https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fmisc%2Fproject_description_file.html&cp%3D2_1_3_11
+
+    if project.get("nature", "") in _ProjectNature.__members__:
+        assert isinstance(project["nature"], str)
+        nature = _ProjectNature[project["nature"]]
+    else:
+        nature = _ProjectNature.project
+
+    return E.projectDescription(
+        E.name(derive_project_name_from_aird(project["location"])),
+        E.comment(),
+        E.projects(),
+        E.buildSpec(),
+        E.natures(E.nature(nature.value)),
+    )
 
 
-def import_project(project: dict[str, str]) -> None:
+def write_updated_project_file(project: _ProjectDict) -> None:
+    project_description_file: pathlib.Path = project["location"] / ".project"
+    project_description_file.write_bytes(etree.tostring(project["etree"]))
+
+
+def import_project(project: _ProjectDict) -> None:
     log.info("Loading project into workspace: %s", str(project["location"]))
     importProject(str(project["location"]))
     log.info("Loading of project was successful: %s", str(project["location"]))
 
 
-if __name__ == "__main__":
+def resolve_duplicate_names(projects: list[_ProjectDict]) -> None:
+    duplicated_projects_grouped = group_projects_by_name(projects)
+
+    for duplicated_projects in duplicated_projects_grouped:
+        for duplicated_project in duplicated_projects:
+            append_revision_to_project_name(duplicated_project)
+
+    # Check again for duplicated projects (same project name and same revision)
+    duplicated_projects_grouped = group_projects_by_name(projects)
+    for duplicated_projects in duplicated_projects_grouped:
+        counter = 0
+        for duplicated_project in duplicated_projects:
+            add_suffix_to_project_name(duplicated_project, str(counter))
+            counter += 1
```

Comment:
```suggestion
    for counter, duplicated_projects in enumerate(duplicated_projects_grouped):
        for duplicated_project in duplicated_projects:
            add_suffix_to_project_name(duplicated_project, str(counter))
```

Reply:
Nice one :)

Reply:
I just had to move the counter to the second loop.